### PR TITLE
Fixes circles not rendering on Mesa 24.1 with AMDGPU

### PR DESCRIPTION
**Pull request by cutephoton**
On MESA 24.1 with AMD gpu, the fragment shader will fail to render circles. No root cause has been definitively identified, but it appears to be a failed optimization of the shaders specific to AMD GPUs on Mesa >=24.1. (Intel IGPU was not impacted. Nvidia untested.)

The change works around the issue by calculating the final blur value in the vertex shader rather than in the fragment shader.

Fixes: https://github.com/maplibre/maplibre-gl-js/issues/4062
See discussion (& images) here: https://github.com/maplibre/maplibre-gl-js/discussions/4402
Verified with `npm run test-render -- circle --open-browser`

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,8 @@
 - Fixed a performance regression related to symbol placement ([#4599](https://github.com/maplibre/maplibre-gl-js/pull/4599))
 - Fix a bug where cloning a Transform instance didn't include the `lngRange`. This caused a bug where
 using `transformCameraUpdate` caused the `maxBounds` to stop working just for east/west bounds. ([#4625](https://github.com/maplibre/maplibre-gl-js/pull/4625))
+- Fix circle won't render on mesa 24.1 with AMD GPU ([#4062](https://github.com/maplibre/maplibre-gl-js/issues/4062))
```
Comment (HarelM) [inline CHANGELOG.md:27]: This needs to be moved upwards as version 4.7.0 was already released.

**Comment by cutephoton** [comment]
Done. It was a fascinating deep dive into something I haven't explored in much depth before. What better way to learn than to start with a mystery... :)

**Review by HarelM — APPROVED** [review]

**Comment by HarelM** [comment]
@cutephoton Can you please help me merge these changes to the globe branch?
I'm not sure I have a good way to test that these changes are still needed and what's the best way to add them.
CC: @kubapelc

**Comment by kubapelc** [comment]
I'll have a look at it and try port the changes to globe branch by hand. I have no way of checking whether they are still needed though.

**Pull request merged**